packages/sogrim-app/src/components/Banner/BannerDialogs/ImportGilion.tsx

```diff
@@ -14,36 +22,42 @@ import { useAuth } from "../../../hooks/useAuth";
 export interface ImportGilionProps {
   handleSkip: () => void;
   handleClose: () => void;
-  handleError: (msg: string) => void;
 }
 
 export const ImportGilion: React.FC<ImportGilionProps> = ({
   handleSkip,
   handleClose,
-  handleError,
 }) => {
   const [ugText, setUgText] = useState<string | null>(null);
+  const [backdropOpen, setBackdropOpen] = useState(false);
   const { userAuthToken } = useAuth();
 
-  const { mutate, isError } = useUpdateUserUgData(userAuthToken);
+  const { mutate, isError, isLoading, isSuccess } =
+    useUpdateUserUgData(userAuthToken);
 
   const handleChangeTextField = (e: React.ChangeEvent<HTMLInputElement>) => {
     e.preventDefault();
     setUgText(e.target.value);
   };
   const handleSend = () => {
     if (ugText) {
+      setBackdropOpen(true);
       mutate(ugText);
     }
-    handleClose();
   };
 
   useEffect(() => {
-    const errorMsg = isError
-      ? "ייבוא גיליון הציונים כשל. האם העתקתם את כל גיליון הציונים (ולא תעודת ציונים!) , דרך דפדפן כרום?"
-      : "";
-    handleError(errorMsg);
-  }, [handleError, isError]);
+    if (isError) {
+      setBackdropOpen(false);
+    }
+  }, [isError]);
+
+  useEffect(() => {
```

Comment:
Why do we need two different ```useEffect()``` and not merge them to one?
useEffect(() => {
if (isError) {
      setBackdropOpen(false);
    }
else if(isSuccess) {
      setBackdropOpen(false);
      handleClose();
}
}, [isError, isSuccess]);